### PR TITLE
Add event types (events without signup and vice versa)

### LINKED ISSUE #5: Add support for events without signup
We will use this system for all events on tietokilta.fi, and some events will not have signup. Therefore Events should have a mechanism where they have no registration.

### LINKED ISSUE #11: Add support for signups without date
We may want to have signups that have no "event" attached. This allows us to e.g. list them as open signups without showing them in a calendar.

Practically, this would be implemented by allowing signups to have no set date. This seems simple but may require changes to the UI, `Event.defaultScope` and possibly the API.

### PR DESCRIPTION
**Pull request by PurkkaKoodari**
Closes #5, closes #11.

This PR adds support for events without a signup and signups without an event date. The choice is implemented with a select box in the event editor: "event with signup", "event without signup" or "signup without event".

This is a moderately small change, since many parts of the code already handled these cases.

- In the event list, signups without an event are shown first and not distinguished otherwise. This may be redesigned later.

- There is a slight UX to code quality tradeoff here. When saving an event without a date or registration, the date or registration start/end fields are cleared. This makes code cleaner (no need to separately check type field(s), less reduncancy of data) but may cause a slight inconvenience to users.

- Events can still be saved as "event + signup" with one of the fields empty, which will convert them to "only event/signup" - this will be fixed in #32.

### CODE REVIEW

File: server/mail/signupConfirmation.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export default async (signup: Signup) => {
   ];
 
   const edited = answers.some((answer) => answer.createdAt.getTime() !== answer.updatedAt.getTime());
-  const date = moment(event.date).tz('Europe/Helsinki').format('DD.MM.YYYY HH:mm');
+  const date = event.date && moment(event.date).tz('Europe/Helsinki').format('DD.MM.YYYY HH:mm');
```
Comment (lakaniemi) [inline server/mail/signupConfirmation.ts:34]: Using `&&` outside of JSX can cause weird types - now this is either moment's date or `false` (so `boolean`). Does not necessarily cause functional difference though as it's still falsy

File: server/services/signup/computeSignupPosition.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ async function sendPromotedFromQueueEmail(signup: Signup, eventId: Event['id'])
 
   const params = {
     event,
-    date: moment(event.date).tz('Europe/Helsinki').format('DD.MM.YYYY HH:mm'),
+    date: event.date && moment(event.date).tz('Europe/Helsinki').format('DD.MM.YYYY HH:mm'),
```
Comment (lakaniemi) [inline server/services/signup/computeSignupPosition.ts:18]: Same as above

File: server/services/signup/createNewSignup.ts [hunk 3]
```diff
@@ -23,16 +22,12 @@ export interface SignupCreateResponse {
...
   }
 
-  const now = moment();
-
-  const isOpened = now.isSameOrAfter(moment(event.registrationStartDate));
-  const isClosed = now.isAfter(moment(event.registrationEndDate));
-
-  return isOpened && !isClosed;
+  const now = new Date();
+  return now >= event.registrationStartDate && now <= event.registrationEndDate;
```
Comment (lakaniemi) [inline server/services/signup/createNewSignup.ts:30]: Seems like there's now both `moment` and JS's own `Date` used in the codebase. Would it make sense to stick to one way to handle dates, for the sake of consistency?

File: src/components/FieldRow/index.tsx [hunk 4]
```diff
@@ -33,7 +35,11 @@ export default function FieldRow<P = unknown>({
   if (children) {
     field = children;
   } else {
-    field = <Field as={as} name={name} required={required} {...props} />;
+    let fieldProps = props;
+    if (checkLabel !== undefined) {
+      fieldProps = { ...props, label: checkLabel };
+    }
+    field = <Field as={as} name={name} required={required} {...fieldProps} />;
```
Comment (lakaniemi) [inline src/components/FieldRow/index.tsx:55]: What exactly is happening here :D Why not just use the label prop?

File: src/modules/editor/types.d.ts [hunk 5]
```diff
@@ -44,8 +44,12 @@ export interface EditorQuota extends AdminEvent.Update.Quota {
   key: Quota.Id;
 }
 
+type EditorEventType = 'event' | 'event+signup' | 'signup';
```
Comment (lakaniemi) [inline src/modules/editor/types.d.ts:47]: Could also be enum, but type is fine too 👍

File: src/components/FieldRow/index.tsx (hunk 4 above)
Comment (PurkkaKoodari) [reply src/components/FieldRow/index.tsx:55]: For normal fields, the `label` prop is placed in the left column by our component. Checkboxes have a label both in the left column and next to the checkbox, and `Form.Check` expects the latter as `label` as well - so we can't just pass it in `...props`.

I think this hack makes more sense than renaming `label` for every other field type, or using a special name for `label` with checkboxes. Will add a comment, though.

File: server/services/signup/createNewSignup.ts (hunk 3 above)
Comment (PurkkaKoodari) [reply server/services/signup/createNewSignup.ts:30]: Definitely. Haven't proceeded with this yet, as there may be subtle stuff with timezones. Maybe also replace both with `Intl` if the support is good enough for us, to cut the bundle size.

File: server/mail/signupConfirmation.ts (hunk 1 above)
Comment (PurkkaKoodari) [reply server/mail/signupConfirmation.ts:34]: According to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND), *the `&&` operator actually returns the value of one of the specified operands*. VSCode also says `string | null`:
image

File: src/modules/editor/types.d.ts (hunk 5 above)
Comment (PurkkaKoodari) [reply src/modules/editor/types.d.ts:47]: Would have preferred an enum here, but I believe they can't be defined in `.d.ts`. As to why the `types` files are `.d.ts` - I tried changing it and it caused some errors related to how the interfaces are defined, which I didn't look into. I think that change will be more meaningful as a separate refactor if desired.

File: server/mail/signupConfirmation.ts (hunk 1 above)
Comment (lakaniemi) [reply server/mail/signupConfirmation.ts:34]: TIL

**Pull request merged**